llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1081,9 +1080,8 @@ static VPActiveLaneMaskPHIRecipe *addVPLaneMaskPhiAndUpdateExitBranch(
       "index.part.next");
 
   // Create the active lane mask instruction in the VPlan preheader.
-  auto *EntryALM =
-      Builder.createNaryOp(VPInstruction::ActiveLaneMask, {EntryIncrement, TC},
-                           DL, "active.lane.mask.entry");
+  auto *EntryALM = Builder.createGetActiveLaneMask(EntryIncrement, TC, DL,
```

Comment:
Is it worth to check function in line 1075:
TripCount = Builder.createNaryOp(VPInstruction::CalculateTripCountMinusVF,
                                     {TC}, DL); 
To see  if that is the reason there are more scale and selects in the llvm-ir.

Reply:
The expansion from VPlan to LLVM IR of instructions like
```
  EMIT vp<%3> = TC > VF ? TC - VF : 0 ir<%n>
  EMIT vp<%4> = VF * Part + ir<0>
```
indeed in some cases may result in dead instructions, since it does not depend and has no knowledge of the logic of expansion of things like
```
EMIT vp<%17> = active lane mask vp<%16>, vp<%3>
```
and I want to keep it that way.